profiler/profiler.go

```diff
@@ -224,12 +224,24 @@ func (o *allowUntilSuccess) do(f func() error) (err error) {
 // additional calls will be ignored.
 func Start(cfg Config, options ...option.ClientOption) error {
 	startError := startOnce.do(func() error {
```

Comment:
If the intention is that someone would re-start the profiler, `startOnce` would prevent that

Reply:
Oh, absolutely. 

I've also added a test for this. Should we implement anything to:

- prevent a double start?
- notify the caller when the profiler actually has stopped (something like the `profilingDone` channel)?

Reply:
I wonder if we should change the `StartWithContext` API and let it return a stop function. I was thinking about something like this (untested and scratched here in the comment box):


```go
func StartWithContext(ctx context.Context, cfg Config, options ...option.ClientOption) (error, func()) {
	ctx, cancel := context.WithCancel(ctx)
	return start(ctx, cfg, options...), func() {
		cancel()
		<-profilingDone
	}
}
```

Reply:
... of course we could enhance it with a named func type and timeout for waiting to read from the channel etc.

Reply:
... maybe the context should be decoupled though so that only the returned stop function can cancel the profiling, so that the user always has to await for the profiling to be finished. The user then would need to forward ctx cancellation ... 

WDYT?